### ISSUE TITLE
How to get a shell into a running chef-container

### ISSUE THREAD

**Issue by snayakie** [chef-boneyard/chef-init#24]
Two questions here:
1. After getting a chef-container built image, and running it as container, I would like to get 'into' it with a shell to look around and to fix things. Is there any way to do this?
2. Also, when debugging the 'knife container docker build' for errors in any recipe, is there any way then start the resulting image as a container and get a shell into it to try the things that were failing, fix them, and then I can go outside, fix my recipe and build the chef container again?

So basically, I want to be able to get a shell into the container for the above scenarios, sometimes when the 'build' is failing, and sometimes even after I have built my image successfully.

**Comment by tduffield** [comment]
The best way to enter a running container is nsenter. There is no mechanism built into chef-init for such a thing. If the chef-client run fails then the image is not saved. You can snapshot the container instance that failed and launch it with /bin/bash if you want.

**Comment by snayakie** [comment]
Found nsenter. Thanks.

**Comment by bpesics** [comment]
should be so mentioned in the documentation ;-)
